Assignment4/javipzv-Task07.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+"""Copia de Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1lz4zFy5ma8GUJ0XHjZJaj-f7ja2yZJhy
+
+**Task 07: Querying RDF(s)**
+"""
+
+# !pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+from rdflib.plugins.sparql import prepareQuery
+ns = Namespace("http://somewhere#")
+
+# TO DO
+
+# RDFLib
+def rec_subclasses(className):
+  for s, p, o in g.triples((None, RDFS.subClassOf, className)):
+    if s:
+      if rec_subclasses(s):
+        return [s] + [rec_subclasses(s)]
```

Comment:
this is very inefficient because you are calling the recursive function twice.
It would be better if you just to  a = rec_subclasses(s) and then 
if a:
  return [s] + a